cmd/declaration_extraction/declaration_extraction.go

```diff
@@ -0,0 +1,323 @@
+package declaration_extraction
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"unicode"
+)
+
+type GlobalDeclaration struct {
+	PackageID   string
+	FileID      string
+	StartOffset int
+	Length      int
+	LineNumber  int
+	Name        string
```

Comment:
Also please change `Name` fields to `VariableName`, and the other structs with name field too